### PR TITLE
enable WASM StorageAdapter to also work in mocked `nodejs`

### PR DESCRIPTION
**Pull request by antbern**
# Description of change

This enables the `StorageAdapter` for `wasm`, and also makes sure it works in `nodejs` whenever the `window.localStorage` global exists 🚀 

## Links to any relevant issues

Be sure to reference any related issues by adding `fixes #issue_number`.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by kristifidani** [comment]
Looks good 👍

File: sdk/src/wallet/storage/adapter/wasm.rs [hunk 1]
```diff
@@ -10,35 +12,82 @@ pub const STORAGE_ID: &str = "Wasm";
...
 /// Wasm storage adapter using the browser local storage
 #[derive(Debug)]
-pub struct WasmAdapter(LocalStorage);
+pub struct WasmAdapter {
+    /// field used just to make it impossible to create a [`WasmAdapter`] without using the `new` function.
+    _ignored: (),
+}
+
+const PROBE_KEY: &str = "iota-sdk-availability-test";
+const PROBE_VALUE: &str = "local storage is available!";
```
Comment (kristifidani) [inline sdk/src/wallet/storage/adapter/wasm.rs]: do we need to keep these as consts or we can put them inside the new() function where they are used

Comment (antbern) [reply sdk/src/wallet/storage/adapter/wasm.rs]: Moved them into the `new` function 💯

**Review by hauju — APPROVED** [review]
Looks good to me!

**Pull request merged**